lib/doorkeeper/request.rb

```diff
@@ -4,30 +4,25 @@ module Doorkeeper
   module Request
     class << self
       def authorization_strategy(response_type)
-        get_strategy(response_type, authorization_response_types)
-      rescue NameError
-        raise Errors::InvalidAuthorizationStrategy
+        build_strategy_class(response_type)
```

Comment:
Why we replaced it? :thinking: 

Also I see that `token_strategy` do a check for available types, but authorization - not.. I'm kind of confused ...

Reply:
Because:
- At authorizations_controller **return error immediately** when pre_auth is not authorized:
https://github.com/doorkeeper-gem/doorkeeper/pull/1277/files#diff-1282d211312f992c2a614d1f4a6302feR94-R97
- `authorization_strategy` is called from `authorizations_controller.rb` **only when pre_auth is passed all the validation check (all parameters is valid)**:
    call `authorization_request`
https://github.com/doorkeeper-gem/doorkeeper/blob/6eda02d7b0e9971a71850aff49737e1cf141d2be/app/controllers/doorkeeper/authorizations_controller.rb#L83-L85
    then call `authorization_strategy ` at https://github.com/doorkeeper-gem/doorkeeper/blob/6eda02d7b0e9971a71850aff49737e1cf141d2be/lib/doorkeeper/server.rb#L12

- So I replaced all the error checking and just do `build_strategy_class`.

Reply:
`token_strategy` is otherwise called from `tokens_controller.rb`, which doesn't have object like `pre_authorization` to check whether the params are valid or not. So we need to check if token strategy is valid at here.

Reply:
It's totally OK, thanks for extended clarification :bowing_man: